pytorch_transformers/tokenization_utils.py

```diff
@@ -666,35 +696,146 @@ def _convert_token_to_id(self, token):
     def encode(self, text, text_pair=None, add_special_tokens=False, **kwargs):
         """
         Converts a string in a sequence of ids (integer), using the tokenizer and vocabulary.
-        
+
         Same as doing ``self.convert_tokens_to_ids(self.tokenize(text))``.
 
         Args:
             text: The first sequence to be encoded.
             text_pair: Optional second sequence to be encoded.
             add_special_tokens: if set to ``True``, the sequences will be encoded with the special tokens relative
                 to their model.
-            **kwargs: passed to the `self.tokenize()` method
         """
         if text_pair is None:
             if add_special_tokens:
-                return self.add_special_tokens_single_sentence(self.convert_tokens_to_ids(self.tokenize(text, **kwargs)))
+                sequence_tokens = self.convert_tokens_to_ids(self.tokenize(text, **kwargs)) if isinstance(text, six.string_types) else text
```

Comment:
Here:
- the `else text` option should be inside the call to `self.convert_tokens_to_ids()`
- we should add the option to supply a list of integer: another `else text` to bypass `self.convert_tokens_to_ids` with a test: `isinstance(text, (list, tuple)) and len(text) > 0 and isinstance(text[0], int)`

Reply:
I was only taking into account the list of integers (token ids), not a list of tokens. Supplying the three formats: `string`, `list[string]` and `list[int]` makes sense, I'm changing this now.